Sources/SourceKitLSP/Swift/DocumentFormatting.swift

```diff
@@ -136,20 +136,44 @@ private func edits(from original: DocumentSnapshot, to edited: String) -> [TextE
 
 extension SwiftLanguageService {
   package func documentFormatting(_ req: DocumentFormattingRequest) async throws -> [TextEdit]? {
-    let snapshot = try documentManager.latestSnapshot(req.textDocument.uri)
+    return try await format(
+      textDocument: req.textDocument,
+      options: req.options
+    )
+  }
+
+  package func documentRangeFormatting(_ req: DocumentRangeFormattingRequest) async throws -> [TextEdit]? {
+    return try await format(
+      textDocument: req.textDocument,
+      options: req.options,
+      extraArgs: [
+        "--offsets",
+        "\(req.range.lowerBound.line):\(req.range.upperBound.line)",
```

Comment:
swift-format expects UTF-8 byte offsets instead of line numbers for `--offsets`. You should be able to get the UTF-8 offset range using

https://github.com/swiftlang/sourcekit-lsp/blob/e73e8b798466aa8161ceb687ad2be092face8176/Sources/SourceKitLSP/Swift/SwiftLanguageService.swift#L1174